test/javascripts/acceptance/chat-quoting-test.js

```diff
@@ -0,0 +1,235 @@
+import { test } from "qunit";
+import { click, currentURL, visit } from "@ember/test-helpers";
+import {
+  acceptance,
+  exists,
+  query,
+  visible,
+} from "discourse/tests/helpers/qunit-helpers";
+import {
+  chatChannels,
+  chatView,
+} from "discourse/plugins/discourse-chat/chat-fixtures";
+import selectKit from "discourse/tests/helpers/select-kit-helper";
+
+let quoteResponse = {
+  bbcode: `[chat quote="martinchat;3875498;2022-02-04T01:12:15Z" channel="The Beam Discussions"]
+  an extremely insightful response :)
+  [/chat]`,
+};
+
+function setupPretenders(server, helper) {
+  server.get("/chat/chat_channels.json", () => helper.response(chatChannels));
+  server.post(`/chat/9/quote.json`, () => helper.response(quoteResponse));
+  server.post(`/chat/4/quote.json`, () => helper.response(quoteResponse));
+  server.post(`/chat/7/quote.json`, () => helper.response(quoteResponse));
+  server.get("/chat/:chatChannelId/messages.json", () =>
+    helper.response(chatView)
+  );
+  server.post("/uploads/lookup-urls", () => {
+    return helper.response([]);
+  });
+}
+
+acceptance("Discourse Chat | quoting out of topic", function (needs) {
+  needs.user({
+    admin: false,
+    moderator: false,
+    username: "eviltrout",
+    id: 1,
+    can_chat: true,
+    has_chat_enabled: true,
+  });
+
+  needs.settings({
+    chat_enabled: true,
+  });
+
+  needs.pretender((server, helper) => {
+    setupPretenders(server, helper);
+  });
+
+  test("it opens the composer and appends the quote", async function (assert) {
```

Comment:
I'm wondering if we should assert that the category is prefilled in the composer too.

Reply:
Ah I did that in one of the other tests, will add it here too:

https://github.com/discourse-org/discourse-topic-chat/blob/09c088eb4a478f1dfa4d4c153eda4ed228188c81/test/javascripts/acceptance/chat-quoting-test.js#L229-L233